app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -125,6 +165,19 @@ private void setUpNavDrawer() {
         }
     }
 
+    private void syncComplete() {
+        downloadProgress.setVisibility(View.GONE);
+        Bus bus = OpenEventApp.getEventBus();
+        bus.post(new RefreshUiEvent());
+        Snackbar.make(mainFrame, getString(R.string.download_complete), Snackbar.LENGTH_SHORT).show();
+    }
+
+    private void downloadFailed() {
+        downloadProgress.setVisibility(View.GONE);
```

Comment:
Counter should reset only in OnCreate() because this is the only place where the whole data download takes place.